### PR TITLE
fix: tighten SYSTEM username lookup to only match computerName$ not just endsWith $

### PR DESCRIPTION
**Pull request by MikeDombo**
**Issue #, if available:**

**Description of changes:**
Update the system name lookup to computerName$ so not any random username ending with $ will work.

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 - [x] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java [hunk 1]
```diff
@@ -287,16 +287,27 @@ public UserPrincipal lookupUserByName(Path path, String name) throws IOException
...
+    private static String getComputerName() {
+        try {
+            return Kernel32Util.getComputerName();
+        } catch (Win32Exception ex) {
+            String env = System.getenv("COMPUTERNAME");
+            if (env != null) {
+                return env;
+            }
+        }
+        return "";
```
Comment (leaf94) [inline src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java:305]: Some details: 
1. In catch block, I think we are trying again with `System.getenv("COMPUTERNAME");` could we add a comment on when do we expect the exception to be thrown? Also should we log.warn it?
2. Could you also add a comment on why we default to empty String and possible implications?

**Review by leaf94 — COMMENTED** [review]
Thank you for super fast and great fix Michael!

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java (hunk 1 above)
Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java]: Updated

**Review by tilo-chen — APPROVED** [review]

**Review by rbattle — APPROVED** [review]
nice comments

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java [hunk 2]
```diff
@@ -287,16 +287,34 @@ public UserPrincipal lookupUserByName(Path path, String name) throws IOException
...
+            String env = System.getenv("COMPUTERNAME");
+            if (env != null) {
+                return env;
+            }
+        }
+        // Default to empty string if we can't find any name because we use getComputerName()+"$", so it needs
+        // to be appendable. The check to see if the user is SYSTEM will return false since just $ won't match the
+        // username, but that is the proper behavior. We do not want to simply fail, instead it will continue to lookup
+        // the user through the regular Windows APIs which will then either work or generate a proper error.
+        return "";
```
Comment (leaf94) [inline src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java:312]: Fantastic!!! I did have the question about why not re-throw up and it answers my question completely.

**Review by leaf94 — APPROVED** [review]

**Pull request merged**